controllers/grid/announcements/AnnouncementTypeGridHandler.php

```diff
@@ -57,15 +60,20 @@ public function __construct()
      */
     public function authorize($request, &$args, $roleAssignments)
     {
-        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
-        $context = $request->getContext();
+        $contextId = $this->getContextId();
+
+        if ($contextId === Application::CONTEXT_ID_NONE) {
+            $this->addPolicy(new PKPSiteAccessPolicy($request, null, $roleAssignments));
+        } else {
+            $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+        }
 
         $announcementTypeId = $request->getUserVar('announcementTypeId');
         if ($announcementTypeId) {
             // Ensure announcement type is valid and for this context
             $announcementTypeDao = DAORegistry::getDAO('AnnouncementTypeDAO'); /** @var AnnouncementTypeDAO $announcementTypeDao */
             $announcementType = $announcementTypeDao->getById($announcementTypeId);
-            if (!$announcementType || $announcementType->getContextId() != $context->getId()) {
+            if (!$announcementType || $announcementType->getContextId() != $contextId) {
```

Comment:
Again, stylistic, but we've been using `?->` and nulls for a handy/readable syntax. Something like:
```
$context?->getId() ?? CONTEXT_ID_NONE
```